src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Range.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Diagnostics;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal sealed partial class LocalRewriter
+    {
+        public override BoundNode VisitRangeExpression(BoundRangeExpression node)
+        {
+            Debug.Assert(node != null);
+
+            if (node.LeftOperand is null)
+            {
+                if (node.RightOperand is null)
+                {
+                    // Case 1:              ".."
+                    // translates to:       Range.All()
+
+                    MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__All);
+                    return MakeCall(node.Syntax, rewrittenReceiver: null, method, ImmutableArray<BoundExpression>.Empty, method.ReturnType);
+                }
+                else
+                {
+                    MethodSymbol method = (MethodSymbol)_compilation.GetWellKnownTypeMember(WellKnownMember.System_Range__ToEnd);
+                    BoundExpression rightOperand = VisitExpression(node.RightOperand);
+                    rightOperand = NullableAlwaysHasValue(rightOperand) ?? rightOperand;
```

Comment:
>NullableAlwaysHasValue [](start = 35, length = 22)

I think in similar situations we also optimize for always null cases.

Reply:
added a prototype comment.

---
In reply to: [185384447](https://github.com/dotnet/roslyn/pull/26405#discussion_r185384447) [](ancestors = 185384447)